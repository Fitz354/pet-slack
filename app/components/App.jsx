import React from 'react';
import ChannelsList from '../components/ChannelsList.jsx';
import Chat from '../components/Chat.jsx';

export default props => (
  <div className="wrapper container-fluid h-100">
    <div className="row h-100">
      <div className="w-25 h-100 pt-3 bg-dark">
        <ChannelsList userName = {props.userName}/>
      </div>
      <div className="col pt-3 pb-5 h-100 bg-light">
        <Chat userName = {props.userName} />
      </div>
    </div>
  </div>
);
