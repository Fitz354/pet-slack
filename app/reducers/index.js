import { handleActions } from 'redux-actions';
import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import * as actions from '../actions';

const messageCreatingState = handleActions({
  [actions.sendMessageRequest]() {
    return 'requested';
  },
  [actions.sendMessageSuccess]() {
    return 'successed';
  },
  [actions.sendMessageFailure]() {
    return 'failed';
  },
}, 'none');

const messages = handleActions({
  [actions.getMessage](state, { payload: { message } }) {
    return [...state, message];
  },
  [actions.sendMessageSuccess](state, { payload: { message } }) {
    return [...state, message];
  },
}, []);

const channels = handleActions({}, []);
const currentChannelId = handleActions({
  [actions.switchChannel](state, { payload: { id } }) {
    return id;
  },
}, 0);

export default combineReducers({
  messageCreatingState,
  channels,
  messages,
  currentChannelId,
  form: formReducer,
});
