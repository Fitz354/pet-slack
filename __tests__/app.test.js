import 'js-polyfills/html';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import timeout from 'timeout-then';
import axios from 'axios';
import nock from 'nock';
import httpAdapter from 'axios/lib/adapters/http';
import { SocketIO, Server } from 'mock-socket';
import _ from 'lodash';

import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import addSocketHandlers from '../app/socket';
import App from '../app/components/App.jsx';
import reducers from '../app/reducers';

Enzyme.configure({ adapter: new Adapter() });

axios.defaults.adapter = httpAdapter;

const getNextId = () => Number(_.uniqueId());

const catchMessage = () => {
  nock('http://localhost')
    .post('/api/v1/channels/1/messages')
    .reply(200, (uri, body) => {
      const data = JSON.parse(body);
      const id = getNextId();
      const attributes = {
        ...data.data.attributes,
        time: '10:00',
        id,
      };

      const response = {
        data: {
          type: 'messages',
          id,
          attributes,
        },
      };

      return response;
    });
};

it('check app', async () => {
  const initState = {
    channels: [
      { id: 1, name: 'general', removable: false },
      { id: 2, name: 'random', removable: false },
    ],
    messages: [],
    currentChannelId: 1,
  };

  const store = createStore(
    reducers,
    initState,
    applyMiddleware(thunk),
  );

  const mockServer = new Server('http://localhost:8080');
  const mockClient = SocketIO('http://localhost:8080');
  addSocketHandlers(mockClient, store.dispatch, 'user1');

  document.body.innerHTML = '<div id="app-container"></div>';
  const wrapper = mount(
    <Provider store={store}>
      <App userName='User1' />
    </Provider>,
    document.getElementById('app-container'),
  );
  expect(wrapper.render()).toMatchSnapshot();

  const messageInput = wrapper.find('.message-input input[type="text"]');
  const messageSubmit = wrapper.find('.message-submit');

  messageInput.simulate('change', { target: { value: 'hey!' } });
  expect(wrapper.render()).toMatchSnapshot();

  catchMessage();
  messageSubmit.simulate('submit');
  await timeout(100);
  expect(wrapper.render()).toMatchSnapshot();

  messageInput.simulate('change', { target: { value: 'how are you?' } });
  catchMessage();
  messageSubmit.simulate('submit');
  await timeout(100);
  expect(wrapper.render()).toMatchSnapshot();

  const newMessageId = getNextId();
  const newMessage = {
    data: {
      type: 'messages',
      id: newMessageId,
      attributes: {
        id: newMessageId,
        userName: 'User2',
        time: '10:02',
        text: 'i\'m fine',
      },
    },
  };
  mockServer.emit('newMessage', newMessage);
  expect(wrapper.render()).toMatchSnapshot();
});
