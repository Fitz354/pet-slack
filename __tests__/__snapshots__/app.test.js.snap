// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check app 1`] = `
<div
  class="wrapper container-fluid h-100"
>
  <div
    class="row h-100"
  >
    <div
      class="w-25 h-100 pt-3 bg-dark"
    >
      <div
        class="channels-list h-100"
      >
        <div
          class="user-name pl-3 pr-3"
        >
          <p
            class="h3 text-white"
          >
            User1
          </p>
        </div>
        <div
          class="nav flex-column nav-pills"
        >
          <a
            class="nav-link text-white active"
            href="#"
          >
            # general
          </a>
          <a
            class="nav-link text-white"
            href="#"
          >
            # random
          </a>
        </div>
      </div>
    </div>
    <div
      class="col pt-3 pb-5 h-100 bg-light"
    >
      <div
        class="chat d-flex flex-column justify-content-between h-100"
      >
        <h2
          class="chat-name"
        >
          #general
        </h2>
        <div
          class="chat-messages col"
        >
          <div
            class="card h-100"
          >
            <div
              class="card-body"
            />
          </div>
        </div>
        <div
          class="chat-input mt-3"
        >
          <form
            class="form-inline"
          >
            <div
              class="form-group w-50 mr-3"
            >
              <input
                class="message-input form-control w-100"
                name="message"
                placeholder="Message to #general"
                type="text"
                value=""
              />
            </div>
            <button
              class="message-submit btn btn-primary"
              disabled=""
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`check app 2`] = `
<div
  class="wrapper container-fluid h-100"
>
  <div
    class="row h-100"
  >
    <div
      class="w-25 h-100 pt-3 bg-dark"
    >
      <div
        class="channels-list h-100"
      >
        <div
          class="user-name pl-3 pr-3"
        >
          <p
            class="h3 text-white"
          >
            User1
          </p>
        </div>
        <div
          class="nav flex-column nav-pills"
        >
          <a
            class="nav-link text-white active"
            href="#"
          >
            # general
          </a>
          <a
            class="nav-link text-white"
            href="#"
          >
            # random
          </a>
        </div>
      </div>
    </div>
    <div
      class="col pt-3 pb-5 h-100 bg-light"
    >
      <div
        class="chat d-flex flex-column justify-content-between h-100"
      >
        <h2
          class="chat-name"
        >
          #general
        </h2>
        <div
          class="chat-messages col"
        >
          <div
            class="card h-100"
          >
            <div
              class="card-body"
            />
          </div>
        </div>
        <div
          class="chat-input mt-3"
        >
          <form
            class="form-inline"
          >
            <div
              class="form-group w-50 mr-3"
            >
              <input
                class="message-input form-control w-100"
                name="message"
                placeholder="Message to #general"
                type="text"
                value="hey!"
              />
            </div>
            <button
              class="message-submit btn btn-primary"
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`check app 3`] = `
<div
  class="wrapper container-fluid h-100"
>
  <div
    class="row h-100"
  >
    <div
      class="w-25 h-100 pt-3 bg-dark"
    >
      <div
        class="channels-list h-100"
      >
        <div
          class="user-name pl-3 pr-3"
        >
          <p
            class="h3 text-white"
          >
            User1
          </p>
        </div>
        <div
          class="nav flex-column nav-pills"
        >
          <a
            class="nav-link text-white active"
            href="#"
          >
            # general
          </a>
          <a
            class="nav-link text-white"
            href="#"
          >
            # random
          </a>
        </div>
      </div>
    </div>
    <div
      class="col pt-3 pb-5 h-100 bg-light"
    >
      <div
        class="chat d-flex flex-column justify-content-between h-100"
      >
        <h2
          class="chat-name"
        >
          #general
        </h2>
        <div
          class="chat-messages col"
        >
          <div
            class="card h-100"
          >
            <div
              class="card-body"
            >
              <div
                class="message"
              >
                <div
                  class="message-header"
                >
                  <span
                    class="message-sender font-weight-bold"
                  >
                    User1 
                  </span>
                  <span
                    class="message-time"
                  >
                    10:00
                  </span>
                </div>
                <p
                  class="pl-3 pr-3"
                >
                  hey!
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chat-input mt-3"
        >
          <form
            class="form-inline"
          >
            <div
              class="form-group w-50 mr-3"
            >
              <input
                class="message-input form-control w-100"
                name="message"
                placeholder="Message to #general"
                type="text"
                value=""
              />
            </div>
            <button
              class="message-submit btn btn-primary"
              disabled=""
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`check app 4`] = `
<div
  class="wrapper container-fluid h-100"
>
  <div
    class="row h-100"
  >
    <div
      class="w-25 h-100 pt-3 bg-dark"
    >
      <div
        class="channels-list h-100"
      >
        <div
          class="user-name pl-3 pr-3"
        >
          <p
            class="h3 text-white"
          >
            User1
          </p>
        </div>
        <div
          class="nav flex-column nav-pills"
        >
          <a
            class="nav-link text-white active"
            href="#"
          >
            # general
          </a>
          <a
            class="nav-link text-white"
            href="#"
          >
            # random
          </a>
        </div>
      </div>
    </div>
    <div
      class="col pt-3 pb-5 h-100 bg-light"
    >
      <div
        class="chat d-flex flex-column justify-content-between h-100"
      >
        <h2
          class="chat-name"
        >
          #general
        </h2>
        <div
          class="chat-messages col"
        >
          <div
            class="card h-100"
          >
            <div
              class="card-body"
            >
              <div
                class="message"
              >
                <div
                  class="message-header"
                >
                  <span
                    class="message-sender font-weight-bold"
                  >
                    User1 
                  </span>
                  <span
                    class="message-time"
                  >
                    10:00
                  </span>
                </div>
                <p
                  class="pl-3 pr-3"
                >
                  hey!
                </p>
              </div>
              <div
                class="message"
              >
                <p
                  class="pl-3 pr-3"
                >
                  how are you?
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chat-input mt-3"
        >
          <form
            class="form-inline"
          >
            <div
              class="form-group w-50 mr-3"
            >
              <input
                class="message-input form-control w-100"
                name="message"
                placeholder="Message to #general"
                type="text"
                value=""
              />
            </div>
            <button
              class="message-submit btn btn-primary"
              disabled=""
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`check app 5`] = `
<div
  class="wrapper container-fluid h-100"
>
  <div
    class="row h-100"
  >
    <div
      class="w-25 h-100 pt-3 bg-dark"
    >
      <div
        class="channels-list h-100"
      >
        <div
          class="user-name pl-3 pr-3"
        >
          <p
            class="h3 text-white"
          >
            User1
          </p>
        </div>
        <div
          class="nav flex-column nav-pills"
        >
          <a
            class="nav-link text-white active"
            href="#"
          >
            # general
          </a>
          <a
            class="nav-link text-white"
            href="#"
          >
            # random
          </a>
        </div>
      </div>
    </div>
    <div
      class="col pt-3 pb-5 h-100 bg-light"
    >
      <div
        class="chat d-flex flex-column justify-content-between h-100"
      >
        <h2
          class="chat-name"
        >
          #general
        </h2>
        <div
          class="chat-messages col"
        >
          <div
            class="card h-100"
          >
            <div
              class="card-body"
            >
              <div
                class="message"
              >
                <div
                  class="message-header"
                >
                  <span
                    class="message-sender font-weight-bold"
                  >
                    User1 
                  </span>
                  <span
                    class="message-time"
                  >
                    10:00
                  </span>
                </div>
                <p
                  class="pl-3 pr-3"
                >
                  hey!
                </p>
              </div>
              <div
                class="message"
              >
                <p
                  class="pl-3 pr-3"
                >
                  how are you?
                </p>
              </div>
              <div
                class="message"
              >
                <div
                  class="message-header"
                >
                  <span
                    class="message-sender font-weight-bold"
                  >
                    User2 
                  </span>
                  <span
                    class="message-time"
                  >
                    10:02
                  </span>
                </div>
                <p
                  class="pl-3 pr-3"
                >
                  i'm fine
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chat-input mt-3"
        >
          <form
            class="form-inline"
          >
            <div
              class="form-group w-50 mr-3"
            >
              <input
                class="message-input form-control w-100"
                name="message"
                placeholder="Message to #general"
                type="text"
                value=""
              />
            </div>
            <button
              class="message-submit btn btn-primary"
              disabled=""
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
